### PR TITLE
Andrew.rodriguez/dd service mapping

### PR DESCRIPTION
**Pull request by zARODz11z**
### What does this PR do?

Documents DD_SERVICE_MAPPING logic
### Motivation

SLS-3501

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/serverless/aws_lambda/configuration.md
Comment (hestonhoffman) [inline content/en/serverless/aws_lambda/configuration.md:336]:
```suggestion
`DD_SERVICE_MAPPING`: This environment variable allows you to remap Datadog [Service][42] names for upstream non-Lambda (inferred spans) services. It accepts a comma-separated list of `old-service|new-service` pairs, delimited by a pipe (`|`), without spaces.
```

File: content/en/serverless/aws_lambda/configuration.md
Comment (hestonhoffman) [inline content/en/serverless/aws_lambda/configuration.md:359]:
```suggestion
If you wish to rename *specific* upstream services, use these keys:
- For API Gateway, use the domain. For example, `r3pmxmplak.execute-api.us-east-2.amazonaws.com|newServiceName`
- For SNS, use the topic ARN. For example,`arn:aws:sns:us-east-1:123456789012:ExampleTopic|newServiceName`
- For SQS, use the queue ARN. For example, `arn:aws:sqs:us-east-1:123456789012:MyQueue|newServiceName`
- For S3, use the bucket ARN. For example, `arn:aws:s3:::example-bucket|newServiceName`
- For eventbridge, use the event.source. For example, `my.event|newServiceName`
- For Kinesis, use the stream ARN. For example, `arn:aws:kinesis:us-east-1:123456789012:stream/MyStream|newServiceName`
- For DynamoDB, use the table ARN. For example, `arn:aws:dynamodb:us-east-1:123456789012:table/ExampleTableWithStream|newServiceName`
- For Lambda URLS, use the domain. For example, `a8hyhsshac.lambda-url.us-east-1.amazonaws.com|newServiceName`
```

File: content/en/serverless/aws_lambda/configuration.md
Comment (hestonhoffman) [inline content/en/serverless/aws_lambda/configuration.md:364]:
```suggestion
   This specifically renames `08se3mvh28.execute-api.eu-west-1.amazonaws.com` to `new-service-name`, providing fine-grained control over upstream service names.
```

File: content/en/serverless/aws_lambda/configuration.md
Comment (hestonhoffman) [inline content/en/serverless/aws_lambda/configuration.md:367]:
```suggestion
   This renames all `lambda_api_gateway` services to `new-service-name`, allowing you to consolidate many services under a single Datadog Service name.
```

File: content/en/serverless/aws_lambda/configuration.md
Comment (hestonhoffman) [inline content/en/serverless/aws_lambda/configuration.md:369]:
```suggestion
For downstream services, refer to `DD_SERVICE_MAPPING` in the [respective tracer's config documentation][41] for your language.
```

**Review by hestonhoffman — APPROVED** [review]
LGTM!

**Comment by zARODz11z** [comment]
> LGTM!

@hestonhoffman great let's hold off merging it for now until my other related prs are merged. I'll reach out if I need someone to hit the button for me.

**Comment by hestonhoffman** [comment]
@zARODz11z No probs! Just give us a shout in #documentation when it's ready to merge.

**Review by tianchu — APPROVED** [review]

**Review by tianchu — APPROVED** [review]

**Pull request merged**
